spring-session/src/main/java/org/springframework/session/IndexResolver.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2014-2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.session;
+
+/**
+ * Strategy interface for resolving the value of the {@link Session}'s index.
+ *
+ * @param <S> the type of Session being handled
+ * @author Vedran Pavic
+ * @since 1.3.0
+ * @see FindByIndexNameSessionRepository
+ */
+public interface IndexResolver<S extends Session> {
+
+	/**
+	 * Resolve the value of the given index from the session.
+	 * @param session the session
+	 * @param indexName the index name to resolve
+	 * @return the resolved index value, {@code null} if none found
+	 */
+	String resolve(S session, String indexName);
```

Comment:
IMO that wouldn't be optimal solution since it implies execution of logic for all supported indexes. This way you extract only what you need in the given situation.